src/Tests/Microsoft.Diagnostics.Monitoring.UnitTestApp/Scenarios/FunctionProbes/FunctionProbesScenario.cs

```diff
@@ -0,0 +1,326 @@
+            {
+                null
+            }, token);
+        }
+
+        private static async Task Test_AsyncMethodAsync(FunctionProbesManager probeManager, PerFunctionProbeProxy probeProxy, CancellationToken token)
+        {
+            MethodInfo method = typeof(StaticTestMethodSignatures).GetMethod(nameof(StaticTestMethodSignatures.AsyncMethod));
+            await RunAsyncTestCaseAsync(probeManager, probeProxy, method, new object[]
+            {
+                5
+            }, token);
+        }
+
+        private static async Task Test_GenericMethodsAsync(FunctionProbesManager probeManager, PerFunctionProbeProxy probeProxy, CancellationToken token)
+        {
+            MethodInfo method = Type.GetType($"{nameof(SampleMethods)}.GenericTestMethodSignatures`2").GetMethod("GenericParameters");
+            Assert.NotNull(method);
+
+            probeProxy.RegisterPerFunctionProbe(method, (object[] actualArgs) => { });
+
+            await WaitForProbeInstallationAsync(probeManager, probeProxy, new[] { method }, token);
+
+            new GenericTestMethodSignatures<bool, int>().GenericParameters(false, 10, "hello world");
+            new GenericTestMethodSignatures<string, object>().GenericParameters("", new object(), 10);
+            new GenericTestMethodSignatures<MyEnum, Uri>().GenericParameters(MyEnum.ValueA, new Uri("https://www.bing.com"), new object());
+
+            Assert.Equal(3, probeProxy.GetProbeInvokeCount(method));
+        }
+
+        public static async Task Test_CaptureValueTypeImplicitThisAsync(FunctionProbesManager probeManager, PerFunctionProbeProxy probeProxy, CancellationToken token)
+        {
+            MethodInfo method = typeof(SampleNestedStruct).GetMethod(nameof(SampleNestedStruct.DoWork));
+            SampleNestedStruct nestedStruct = new();
+            await RunTestCaseAsync(probeManager, probeProxy, method, new object[]
+            {
+                5
+            }, nestedStruct, capturedThisObj: null, token);
+        }
+
+        private static async Task Test_CaptureUnsupportedParametersAsync(FunctionProbesManager probeManager, PerFunctionProbeProxy probeProxy, CancellationToken token)
+        {
+            MethodInfo method = typeof(StaticTestMethodSignatures).GetMethod(nameof(StaticTestMethodSignatures.RefParam));
+
+            await RunTestCaseCoreAsync(probeManager, probeProxy, method, () =>
+            {
+                int i = 10;
+                StaticTestMethodSignatures.RefParam(ref i);
+
+                return Task.CompletedTask;
+            },
+            new object[]
+            {
+                null
+            },
+            hasImplicitThis: false, capturedThisObj: null, token);
+        }
+
+        private static Task RunAsyncTestCaseAsync(FunctionProbesManager probeManager, PerFunctionProbeProxy probeProxy, MethodInfo method, object[] args, object thisObj, CancellationToken token)
```

Comment:
The two overloads only differ in whether the `MethodInfo` is a static method or not. Could you rename these methods to be more clear as to which requires a "this" parameters e.g. `RunStaticMethodTestCaseAsync` and `RunInstanceMethodTestCaseAsync`? Also, please validate that `method.IsStatic` is correct for each method to help test authors know when they used the wrong test case runner with their method.